client/src/App.tsx

```diff
@@ -84,8 +85,9 @@ function App() {
 	const [isRoomError, setIsRoomError] = useState(false);
 
 	const firebaseContext = useContext(FirebaseContext);
-	const [isModalOpen, setIsModalOpen] = useState(false);
+	//const [isModalOpen, setIsModalOpen] = useState(false);
 	const [isPanelOpen, setIsPanelOpen] = useState(true);
+	const [modalState, setModalState] = useState<'new-room' | 'enter-room' | 'none' >('none');
```

Comment:
this is good, however I prefer we use `null` or `undefined` instead of string values like 'none' for clarity. For instance, 'none' could also be 'empty', 'blank', etc, where there is only 1 null or undefined. Something like

`useState<'new-room' | 'enter-room' | null>...`